scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py

```diff
@@ -321,8 +321,14 @@ def update_bgsub_status(self, ws_name, status):
             self._bg_params[ws_name][0] = status
 
     def estimate_background(self, ws_name, niter, xwindow, doSGfilter):
-        ws_bg = EnggEstimateFocussedBackground(InputWorkspace=ws_name, OutputWorkspace=ws_name + "_bg",
-                                               NIterations=niter, XWindow=xwindow, ApplyFilterSG=doSGfilter)
+        try:
+            ws_bg = EnggEstimateFocussedBackground(InputWorkspace=ws_name, OutputWorkspace=ws_name + "_bg",
+                                                   NIterations=niter, XWindow=xwindow, ApplyFilterSG=doSGfilter)
+        except (ValueError, RuntimeError) as e:
+            # ValueError when Niter not positive integer, RuntimeError when Window too small
+            logger.error("Error on arguments supplied to EnggEstimateFocusedBackground: " + str(e))
+            ws_bg = SetUncertainties(InputWorkspace=ws_name, OutputWorkspace=ws_name)  # copy data and zero errors
```

Comment:
I've just retested. The error message from the NIter column now gets logged.

If I then close the UI down and clear the workspaces from the ADS and repeat the test I get a much more verbose error second time. It seems like it originates from this line where SetUncertainties is called. This is triggering the replace handle on the ADS observer and that in turn is creating an exception. Maybe this can be fixed by changing the OutputWorkspace parameter so the original workspace isn't overwritten? Perhaps this should be ws_name_bg?

Reply:
Good spot - yes I didn't indent to replace the original workspace! It should be ws_bg - I'll change that now